sherpa/fit.py

```diff
@@ -1000,23 +998,22 @@ class Fit(NoNewAttributesAfterInit):
 
     Parameters
     ----------
-    data : sherpa.data.Data or sherpa.data.DataSimulFit instance
+    data : `sherpa.data.Data` or `sherpa.data.DataSimulFit`
        The data to be fit.
-    model : sherpa.models.model.Model or sherpa.models.model.SimulFitModel instance
-       The model to fit to the data. It should match the `data` parameter
-       (i.e. be a SimulFitModel object when data is a DataSimulFit).
-    stat : sherpa.stats.Stat instance or None, optional
+    model : `sherpa.models.model.Model` or `sherpa.models.model.SimulFitModel`
+       The model to fit to the data. It should match the ``data`` parameter
+       (i.e. be a `SimulFitModel` object when data is a `DataSimulFit`).
+    stat : `sherpa.stats.Stat` or `None`, optional
        The statistic object to use. If not given then
-       :py:class:`~sherpa.stats.Chi2Gehrels` is used.
+       `Chi2Gehrels` is used.
     method : sherpa.optmethods.OptMethod instance or None, optional
```

Comment:
Looking at https://sherpa--1268.org.readthedocs.build/en/1268/fit/api/sherpa.fit.Fit.html I was surprised that the :look" changed at the `method` parameter from having links to just italicised text, but then I realized that you haven't added the "``` around the `sherpa.optmethods.xxx`, `None`, and `sherpa.estmethod.xxx` items

Reply:
"None" is in fact one of those special words that is picked up for a link even without any formatting. Even though it's formattted like normal text (italics here), it is a link. We can have is "special" by adding backticks. The question is: Do we want that?
That's something I'd like to hear about in this review.

As for the `sherpa.optmethod.xxx`, I simple missed those because they were not links before (I searched for ":py:" to spot the locations that I needed to change), but of course they should become links now! I think the most efficient way to tackle this PR is to go through a bit more carefully once we've decided on the remaining open questions (see above: How should NONE be formatted?). However, we are inconsistent now, and if I miss a few instances tat could be turned into links right now, that OK. This PR will be an improvement in any case and we can fix anytihng I missed at a later point or next time we work on that docstring anyway.

Reply:
In this PR we (aka you ;-) do **not** have to fix every issue in fit.py (or whatever you want to use as the example) to make it worth while to merge.

Reply:
For the `None` vs None vs .. question, what does astropy do? Or are we different enough it's not worth aiming for consistency on the small issues like this?

Reply:
I think we are different enough. For one, they use numpydoc instead of napoleon and their own doc theme.